chromadb/test/property/test_add.py

```diff
@@ -269,12 +270,13 @@ def test_out_of_order_ids(client: ClientAPI) -> None:
     coll = client.create_collection(
         "test", embedding_function=lambda input: [[1, 2, 3] for _ in input]  # type: ignore
     )
-    embeddings: Embeddings = [[1, 2, 3] for _ in ooo_ids]
+    embeddings: Embeddings = [np.array([1, 2, 3]) for _ in ooo_ids]
     coll.add(ids=ooo_ids, embeddings=embeddings)
     get_ids = coll.get(ids=ooo_ids)["ids"]
     assert get_ids == ooo_ids
 
 
+@pytest.mark.xfail(reason="Partial records aren't in our API contract...")
```

Comment:
I don't think we actually intend to support this? This is very old code and I'm not sure it's used anywhere.

Reply:
+1. I don't think this is a pattern we should support.

Reply:
Wait - why would we not support this? What if I have 10 records, one with no Metadata, and 9 with - and I want to add them in one call to add?

Reply:
Partial metadata is supported, should we support partial documents also? Partial embeddings?

Reply:
I think anything that is valid as a single input, should be valid when part of a list

Reply:
This isn't what our types specify which also is a problem.

Reply:
Thinking about this some more, there are weird behaviors possible. For example, if we allow for any entry to be `None`, then it's possible that we get passed an `add` where the nth item contains only metadata, with `None` embedding and `None` document / image / uri. 
This also makes the validation of what's embeddable more difficult, and also makes it harder to pass the right things to the embedding function - we don't strip `None` for example, if we need to embed some documents but not others for which we have embeddings.

I think we should disallow this behavior.

Reply:
Discussed offline - we should allow for interleaved `None` in metadata, but not otherwise.

Reply:
Some examples

Docs can be partial iff embeddings are supplied
coll.add(ids=[1,2,3], metadata=[{k:v}, {k:v}, {k:v}], documents=[None, "a", "b"], embeddings=[[1,2,3], [1,2,3], [1,2,3]] ✅ 

coll.add(ids=[1,2,3], metadata=[{k:v}, {k:v}, {k:v}], documents=[None, "a", "b"], embeddings=None ❌

Metadata can always be partial
coll.add(ids=[1,2,3], metadata=[None, {k:v}, {k:v}], documents=["d", "a", "b"], embeddings=[[1,2,3], [1,2,3], [1,2,3]] ✅

Embeddings can never be partial
coll.add(ids=[1,2,3], metadata=[{k:v}, {k:v}, {k:v}], documents=["d", "a", "b"], embeddings=[None, [1,2,3], [1,2,3]] ❌